### ISSUE TITLE
Assertion failed after completion of migration

### ISSUE THREAD

**Issue by rb-de0** [vapor/core#120]
Application crashed with ```NIO-ELT-#0 (2): Assertion failed``` in following place after completion of migration.

https://github.com/vapor/core/tree/3.0.1/Sources/Async/QueueHandler.swift#L95

My code just changed the database in api-template to MySQL.

I do not know which package's problem, so I submitted it as a Core's issue.

My environment is as follows.

### OS
```
macOS High Sierra 10.13.2
```

### MySQL Server
```
mysql  Ver 14.14 Distrib 5.7.18, for osx10.12 (x86_64) using  EditLine wrapper
```

### Package.resolved
```
{
  "object": {
    "pins": [
      {
        "package": "Console",
        "repositoryURL": "https://github.com/vapor/console.git",
        "state": {
          "branch": null,
          "revision": "273cf2ed1f4daba88a452494c342020171d8c7b8",
          "version": "3.0.0-rc.2.2"
        }
      },
      {
        "package": "Core",
        "repositoryURL": "https://github.com/vapor/core.git",
        "state": {
          "branch": null,
          "revision": "225aff2b69049f2a4d9a4aae71d27ba7fb083032",
          "version": "3.0.1"
        }
      },
      {
        "package": "Crypto",
        "repositoryURL": "https://github.com/vapor/crypto.git",
        "state": {
          "branch": null,
          "revision": "e77aa181580b466fb26865918b62973d9fbfaed6",
          "version": "3.0.0"
        }
      },
      {
        "package": "DatabaseKit",
        "repositoryURL": "https://github.com/vapor/database-kit.git",
        "state": {
          "branch": null,
          "revision": "e594e658cc001e04b8d4ad13881d8714c510f94f",
          "version": "1.0.0-rc.2.2.2"
        }
      },
      {
        "package": "Engine",
        "repositoryURL": "https://github.com/vapor/engine.git",
        "state": {
          "branch": null,
          "revision": "4846ea9abeb04863f82f618ee8cb22650a82ae3c",
          "version": "3.0.0-rc.2.1.1"
        }
      },
      {
        "package": "Fluent",
        "repositoryURL": "https://github.com/vapor/fluent.git",
        "state": {
          "branch": null,
          "revision": "eb63485dfd6fbd0773a25e1b94830fbcedaa8266",
          "version": "3.0.0-rc.2.3.1"
        }
      },
      {
        "package": "FluentMySQL",
        "repositoryURL": "https://github.com/vapor/fluent-mysql.git",
        "state": {
          "branch": null,
          "revision": "7cf4f25b5b8fd22b373944750c6c50b6ddbc8024",
          "version": "3.0.0-rc.2.3"
        }
      },
      {
        "package": "MySQL",
        "repositoryURL": "https://github.com/vapor/mysql.git",
        "state": {
          "branch": null,
          "revision": "198576a6963973a691baaafdb1a01a078affe65c",
          "version": "3.0.0-rc.2.1.2"
        }
      },
      {
        "package": "Routing",
        "repositoryURL": "https://github.com/vapor/routing.git",
        "state": {
          "branch": null,
          "revision": "2fc1d4de22a54848b35ad17b3e7f7816f19ebf90",
          "version": "3.0.0-rc.2"
        }
      },
      {
        "package": "Service",
        "repositoryURL": "https://github.com/vapor/service.git",
        "state": {
          "branch": null,
          "revision": "281a70b69783891900be31a9e70051b6fe19e146",
          "version": "1.0.0"
        }
      },
      {
        "package": "swift-nio",
        "repositoryURL": "https://github.com/apple/swift-nio.git",
        "state": {
          "branch": null,
          "revision": "a0b7f646fba61402da384b3202f54c27debb5e13",
          "version": "1.3.1"
        }
      },
      {
        "package": "swift-nio-ssl",
        "repositoryURL": "https://github.com/apple/swift-nio-ssl.git",
        "state": {
          "branch": null,
          "revision": "ea006b6368dbd9dbfd297deb6ddb3f070b72d043",
          "version": "1.0.1"
        }
      },
      {
        "package": "swift-nio-ssl-support",
        "repositoryURL": "https://github.com/apple/swift-nio-ssl-support.git",
        "state": {
          "branch": null,
          "revision": "c02eec4e0e6d351cd092938cf44195a8e669f555",
          "version": "1.0.0"
        }
      },
      {
        "package": "swift-nio-zlib-support",
        "repositoryURL": "https://github.com/apple/swift-nio-zlib-support.git",
        "state": {
          "branch": null,
          "revision": "37760e9a52030bb9011972c5213c3350fa9d41fd",
          "version": "1.0.0"
        }
      },
      {
        "package": "TemplateKit",
        "repositoryURL": "https://github.com/vapor/template-kit.git",
        "state": {
          "branch": null,
          "revision": "61ed3d3b0df4c181d6b88e06b418f425978ac2af",
          "version": "1.0.0-rc.2.0.3"
        }
      },
      {
        "package": "Validation",
        "repositoryURL": "https://github.com/vapor/validation.git",
        "state": {
          "branch": null,
          "revision": "1ce87fc2d18a8f15a491805825063c8db493a51e",
          "version": "2.0.0-rc.2.1.1"
        }
      },
      {
        "package": "Vapor",
        "repositoryURL": "https://github.com/vapor/vapor.git",
        "state": {
          "branch": null,
          "revision": "63d758beb98efc08e8db960372c64ae3704c7616",
          "version": "3.0.0-rc.2.2.3"
        }
      }
    ]
  },
  "version": 1
}

```

**Comment by rb-de0** [comment]
I think that this issue is related to following PR.
https://github.com/vapor/core/pull/118

**Comment by tanner0101** [comment]
Merged #118. This will be available in the 3.0.2 release. Thanks!
